index.js

```diff
@@ -181,3 +185,55 @@ AddonManager.addInstallListener({
     app.send('addon-install:download-failed', formatInstallData(install));
   }
 });
+
+function pingServer(title, addon) {
+  cookieManager.getCookiesFromHost('ideatown.dev', function(cookies) {
+    let csrf;
+    let cookieStr = '';
+    cookies.forEach(function(c) {
+      cookieStr += c.name + '=' + c.value + ' ';
+      if (c.name === 'csrftoken') csrf = c.value;
+    });
+
+    if (csrf) {
+      getMe(csrf, cookieStr, function(id) {
+        request({
+          url: prefs.METRICS_URL,
+          content: formatEvent(title, id, addon),
+          headers: {
+            'Cookie': cookieStr,
+            'X-CSRFToken': csrf
+          },
+          onComplete: function(response) {
+            console.error(response.text);
+          }
+        }).post();
+      });
+    } else app.send('addon-self:needs-auth');
```

Comment:
I think we can get by without adding this event.

If the user doesn't have a session, then the `sessionid` cookie, not the `csrftoken` cookie, will be missing. But in that case, the user can't be logged into the website, so there's not going to be any JS code listening for these `addon-self` events.

We can modify the django settings so that the session lasts forever by default; I've filed https://github.com/mozilla/idea-town/issues/103 to make that change on the server.

So, just assume you've always got a valid session. That should simplify this code a little bit.